test/long/org/apache/cassandra/hints/HintsWriteThenReadTest.java

```diff
@@ -129,10 +130,10 @@ private void verifyHints(File directory, HintsDescriptor descriptor)
 
                     Row row = mutation.getPartitionUpdates().iterator().next().iterator().next();
                     assertEquals(1, Iterables.size(row.cells()));
-                    assertEquals(bytes(index), row.clustering().get(0));
+                    assertTrue(Arrays.equals(bytes(index).array(), (byte[]) row.clustering().get(0)));
```

Comment:
This test was broken by the changes in CASSANDRA-15393 which introduced the `ArrayClustering` type .
If for some reason we switch to another Clustering that use ByteBuffers the test will fail again.
Should we not use  something like:

```
assertEquals(bytes(index), toByteBuffer(row.clustering().get(0)));
...
private ByteBuffer toByteBuffer(Object obj)
{
    if (obj instanceof ByteBuffer)
       return (ByteBuffer) obj;

    return ByteBuffer.wrap((byte[]) obj);
}
```

Reply:
sgtm +1